packages/firestore/package.json

```diff
@@ -33,31 +29,32 @@
     "test:lite:prod": "node ./scripts/run-tests.js --platform node_lite --main=lite/index.ts 'test/lite/**/*.test.ts'",
     "test:lite:browser": "karma start --single-run --lite",
     "test:lite:browser:debug": "karma start --browsers=Chrome --lite --auto-watch",
+    "pretest": "yarn test:prepare",
```

Comment:
Unlike the database PR, all tests including the compat tests remain in the main `firestore` package. I tried to move compat tests to `firestore-compat`, but it got really messy and requires exporting a lot of internal symbols from the main package.

`pretest` copies source code from `firestore-compat` to the `compat` folder in the main package, and replaces import path from `@firebase/firestore` to directly point to source code in the main package. This recreates the code structure before the split, so tests run correctly.

@schmidt-sebastian

Reply:
However there are 2 issues I haven't figure out, and need your help:

- `test:node:prod` doesn't run
- There are 2 failing tests when running `test:browser`

Thanks!

Reply:
I will take a lot this morning.